src/FactoryCity.java

```diff
@@ -0,0 +1,28 @@
+import java.util.ArrayList;
+
+public class FactoryCity implements Factory {
+	private static FactoryCity fc;
+
+	private FactoryCity() {
+		
+	}
+	
+	public static FactoryCity getFC() {
+		if (fc == null) {
+			fc = new FactoryCity();
+		}
+		return fc;
+	}
+	
+	@Override
+	public void createCity(String name, int cityX, int cityY) {
+		Blackboard bb = Blackboard.createInstance();
+		ArrayList<City> cities = bb.getCities();
```

Comment:
The database should be calling the factory, not the other way around. This function only needs to instantiate + return a city with the parameters.